src/emit/soroban/target.rs

```diff
@@ -43,7 +43,18 @@ impl<'a> TargetRuntime<'a> for SorobanTarget {
         slot: &mut IntValue<'a>,
         function: FunctionValue<'a>,
         ns: &ast::Namespace,
+        storage_type: &Option<StorageType>,
     ) -> BasicValueEnum<'a> {
+        let storage_type = if let Some(storage_type) = storage_type {
```

Comment:
To reduce code duplication and improve maintainability, consider extracting the mapping of StorageType to integer values into a helper function. This will ensure consistency across different functions and make future updates easier.

Both storage_load and storage_store functions contain similar code for mapping StorageType.

```rust
fn map_storage_type(storage_type: Option<&StorageType>) -> u64 {
    if let Some(storage_type) = storage_type {
        match storage_type {
            StorageType::Persistent(_) => 0,
            StorageType::Temporary(_) => 1,
            StorageType::Instance(_) => 2,
        }
    } else {
        0 // Default to Persistent
    }
}
```

Then, in your functions, you can use:

```rust
let storage_type = map_storage_type(storage_type);
```